common/crypto/openssl/cert.c

```diff
@@ -1013,3 +1028,396 @@ oe_result_t oe_cert_get_validity_dates(
 
     return result;
 }
+
+#ifdef OE_BUILD_ENCLAVE
+/*
+ * Parse the name string into X509_NAME struct. The format of the string is
+ * "KEY1=VALUE1,KEY2=VALUE2,KEY3=VALUE3...".
+ */
+static X509_NAME* _X509_parse_name(const char* name_string)
+{
+    char* key = NULL;
+    char* value = NULL;
+    char* copy = NULL;
+    char* tmp = NULL;
+    X509_NAME* name = NULL;
+    int error = 1;
+
+    name = X509_NAME_new();
+    if (name == NULL)
+        goto done;
+
+    copy = strdup(name_string);
+    if (!copy)
+        goto done;
+
+    tmp = copy;
+    key = strsep(&tmp, "=");
+    while (key != NULL)
+    {
+        value = strsep(&tmp, ",");
+
+        if (!value)
+            goto done;
+
+        if (!X509_NAME_add_entry_by_txt(
+                name, key, MBSTRING_ASC, (unsigned char*)value, -1, -1, 0))
+            goto done;
+
+        key = strsep(&tmp, "=");
+    }
+
+    error = 0;
+
+done:
+    if (copy)
+        free(copy);
+    if (error && name)
+    {
+        X509_NAME_free(name);
+        name = NULL;
+    }
+
+    return name;
+}
+
+/*
+ * Decode the OID from its BER format (a hex string) into a dot-notation string
+ */
+static char* _decode_oid_to_str(char* oid, int oid_size)
```

Comment:
This is problematic ... any time we introduce parsing of new arbitrary binary data structures, there's a potential for vulnerabilities. I definitely prefer if we could spend some more time exploring additional implementation options here, possibly as a follow-up PR. 

One option may be to reuse the ASN handling code from mbedtls; their source code is nicely self-contained and partitioned, and I believe we've done this on the Windows side to get some parity behavior with OpenSSL.

Reply:
Removed this function.